pip/pip-298.md

```diff
@@ -0,0 +1,105 @@
+
+## Motivation
+
+Currently, Pulsar transactions do not have configurable isolation levels. By introducing isolation level configuration for consumers, we can enhance the flexibility of Pulsar transactions.
+
+## Goal
+
+### In Scope
+
+Implement Read Committed and Read Uncommitted isolation levels for Pulsar transactions.Allow consumers to configure isolation levels during the building process.
+
+### Out of Scope
+
+None.
+
+## API Changes
+
+Update the PulsarConsumer builder process to include isolation level configurations for Read Committed and Read Uncommitted.
+
+### Before the Change
+
+The PulsarConsumer builder process currently does not include isolation level configurations. The consumer creation process might look like this:
+
+```
+PulsarClient client = PulsarClient.builder().serviceUrl("pulsar://localhost:6650").build();
+
+Consumer<String> consumer = client.newConsumer(Schema.STRING)
+        .topic("persistent://my-tenant/my-namespace/my-topic")
+        .subscriptionName("my-subscription")
+        .subscriptionType(SubscriptionType.Shared)
+        .subscribe();
+```
+
+### After the Change
+
+Update the PulsarConsumer builder process to include isolation level configurations for Read Committed and Read Uncommitted. Introduce a new method isolationLevel() in the consumer builder, which accepts an enumeration value representing the isolation level:
+
+```
+public enum IsolationLevel {
+    READ_COMMITTED,
+    READ_UNCOMMITTED
+}
+```
+
+Then, modify the consumer creation process to include the new isolation level configuration:
+
+```
+PulsarClient client = PulsarClient.builder().serviceUrl("pulsar://localhost:6650").build();
+
+Consumer<String> consumer = client.newConsumer(Schema.STRING)
+        .topic("persistent://my-tenant/my-namespace/my-topic")
+        .subscriptionName("my-subscription")
+        .subscriptionType(SubscriptionType.Shared)
+        .subscriptionIsolationLevel(IsolationLevel.READ_COMMITTED) // Adding the isolation level configuration
+        .subscribe();
+```
+
+With this change, users can now choose between Read Committed and Read Uncommitted isolation levels when creating a new
+consumer. If the isolationLevel() method is not called during the builder process, the default isolation level will be
+Read Committed.
+Note that this is a subscription dimension configuration, and all consumers under the same subscription need to be
+configured with the same IsolationLevel.
+
+## Implementation
+
+### Client Changes
```

Comment:
Just very small note. In the template of the pip we have a section "## Public-facing Changes" and several h3 heading so users can clearly see what's changing in any public facing API. In your case I presume if the binary protocol changes.